Documentation: Viewer role can arbitrarily query data sources directly

MattDavis00 opened the pull request:
### What is this feature?

This pull request adds documentation making it clear that `Viewer's can arbitrarily query data sources directly`, depite not being able to edit the dashboard. They can run any arbitrary query against the data source.

### Why do we need this feature?

In the current default configuration & implementation, a Viewer can actually execute arbitrary queries against any data source attached to their organisation.

I don't believe the docs make this clear at all on the roles & permissions index page, and hence is a large oversight in terms of the security model assumptions people are using when configuring Grafana.

Anybody who has assumed a view-only dashboard is a restricted frontend to a data source, like myself, are incorrect. While the dashboard, panels and underlying queries are static and immutable; you can actually use the API to execute any query you like. An authenticated Viewer can execute any query they please, mallicious or not.

### Who is this feature for?

Any Admin or Editor looking at creating dashboard but incorrectly assuming that the queries they create are immutable to Viewers. Viewers can instead query any data source within their organisation directly, bypassing the restricted view-only dashboard.

### Which issue(s) does this PR fix?

This documentation highlights the issues mentioned here:

https://github.com/grafana/grafana/discussions/43487
https://github.com/grafana/grafana/issues/30552
https://github.com/grafana/grafana/issues/26567
https://github.com/grafana/grafana/issues/18157
https://github.com/grafana/grafana/issues/31847

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=69590) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=69590) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=69590) it.</sub>

MattDavis00 commented:
Bump

MattDavis00 commented:
Bump for inactivity - this is still an issue

MattDavis00 commented:
Bump - still an issue.

bergquist commented:
@Jguer any objections to merging this?

bergquist reviewed (approved):
LGTM

@GrafanaWriter did this slip or are we waiting for an response from engineering?

MattDavis00 committed:
Add docs that Viewer can query data source directly

In the current default configuration & implementation, a Viewer
can actually execute arbitrary queries against any data source
attached to their organisation.

I don't believe the docs make this clear at all on the roles &
permissions index page, and hence is a large oversight in terms
the security model people are using when configuring Grafana.

Anybody who has relied on a data source query being static and
immutable provided the dashboard is non-editable is incorrect.
An authenticated Viewer can just execute any query they like.

Signed-off-by: Matt Davis <mw.davis@hotmail.co.uk>

MattDavis00 commented:
Rebased and merge conflicts fixed - Matt

Jguer commented:
Hey @MattDavis00, thanks for this addition to the documentation, I'll merge it in.
 
We're looking for other ways of making this information more visible and encourage people to use mechanisms such as oauth forwarding with datasources that support it to ensure viewers have the strictest access necessary. 

If you have any suggestions, please create issues for them (preferably prefaced with `Auth:` so we can find it better). We'd be very grateful. Once again, thanks for your contribution
